File: backend/src/main/kotlin/com/example/backend/controllers/TicketCaisseController.kt
package com.example.backend.controllers

import com.example.backend.models.TicketCaisse
import com.example.backend.services.TicketCaisseService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/admin/tickets")
class TicketCaisseController(
    private val ticketCaisseService: TicketCaisseService
) {

    @GetMapping
    fun getAllTickets(): ResponseEntity<List<TicketCaisse>> {
        return ResponseEntity.ok(ticketCaisseService.getAllTickets())
    }

    @GetMapping("/{id}")
    fun getTicketById(@PathVariable id: Int): ResponseEntity<TicketCaisse> {
        return ResponseEntity.ok(ticketCaisseService.getTicketById(id))
    }

    @GetMapping("/codebarre/{codebarre}")
    fun getTicketByCodebarre(@PathVariable codebarre: Int): ResponseEntity<TicketCaisse?> {
        return ResponseEntity.ok(ticketCaisseService.getTicketByCodebarre(codebarre))
    }

    @PostMapping
    fun createTicket(@RequestBody ticket: TicketCaisse): ResponseEntity<TicketCaisse> {
        return ResponseEntity.ok(ticketCaisseService.createTicket(ticket))
    }

    @PutMapping("/{id}")
    fun updateTicket(@PathVariable id: Int, @RequestBody updatedTicket: TicketCaisse): ResponseEntity<TicketCaisse> {
        return ResponseEntity.ok(ticketCaisseService.updateTicket(id, updatedTicket))
    }

    @DeleteMapping("/{id}")
    fun deleteTicket(@PathVariable id: Int): ResponseEntity<Void> {
        ticketCaisseService.deleteTicket(id)
        return ResponseEntity.noContent().build()
    }
}

File: backend_py/pharmaPython/app/api/v1/ticket_caisse_controller.py
from fastapi import APIRouter, Depends, HTTPException, Path
from sqlalchemy.orm import Session
from typing import List

from app.api.deps import get_db
from app.models.ticket_caisse import TicketCaisse, TicketCaisseSchema, TicketCaisseIn
from app.services.ticket_service import TicketService

router = APIRouter(prefix="/admin/tickets", tags=["Tickets"])

@router.get("/", response_model=List[TicketCaisseSchema])
def get_all_tickets(db: Session = Depends(get_db)):
  return TicketService(db, ticket_repo=...).get_all_tickets()

@router.get("/{id}", response_model=TicketCaisseSchema)
def get_ticket_by_id(id: int = Path(..., ge=1), db: Session = Depends(get_db)):
  t = TicketService(db, ticket_repo=...).get_ticket_by_id(id)
  if not t:
    raise HTTPException(status_code=404, detail="Ticket non trouvé")
  return t

@router.get("/codebarre/{codebarre}", response_model=TicketCaisseSchema | None)
def get_ticket_by_codebarre(codebarre: int, db: Session = Depends(get_db)):
  return TicketService(db, ticket_repo=...).get_ticket_by_codebarre(codebarre)

@router.post("/", response_model=TicketCaisseSchema, status_code=201)
def create_ticket(ticket: TicketCaisseIn, db: Session = Depends(get_db)):
  return TicketService(db, ticket_repo=...).create_ticket(ticket)

@router.put("/{id}", response_model=TicketCaisseSchema)
def update_ticket(id: int, updated_ticket: TicketCaisseIn, db: Session = Depends(get_db)):
  return TicketService(db, ticket_repo=...).update_ticket(id, updated_ticket)

@router.delete("/{id}", status_code=204)
def delete_ticket(id: int, db: Session = Depends(get_db)):
  TicketService(db, ticket_repo=...).delete_ticket(id)
  return {"message": "deleted"}
